src/pages/memos/[functionId]/MemosForm.js

```diff
@@ -0,0 +1,531 @@
+      }
+
+      try {
+        const res = await getRequest({
+          extension: FinancialRepository.FiMemo.get,
+          parameters: `_recordId=${response.recordId}`
+        })
+
+        formik.setFieldValue('reference', res.record.reference)
+      } catch (error) {}
+      invalidate()
+    }
+  })
+  const editMode = !!formik.values.recordId || !!recordId
+
+  useEffect(() => {
+    ;(async function () {
+      try {
+        const res = await getRequest({
+          extension: SystemRepository.Defaults.qry,
+          parameters: `_filter=vatPct`
+        })
+
+        const vatPctValue = res.list.find(item => item.key === 'vatPct')?.value || '0'
+        setInitialVatPct(vatPctValue)
+      } catch (exception) {}
+    })()
+  }, [])
+  useEffect(() => {
+    let calculatedAmount = parseInt(formik.values.subtotal)
+
+    if (formik.values.isSubjectToVAT) {
+      formik.setFieldValue('vatPct', initialVatPct)
+      const vatAmount = formik.values.subtotal / initialVatPct
+      formik.setFieldValue('vatAmount', vatAmount)
+
+      calculatedAmount += vatAmount
+    } else {
+      formik.setFieldValue('vatPct', '')
+      formik.setFieldValue('vatAmount', 0)
+    }
+
+    formik.setFieldValue('amount', calculatedAmount)
+    formik.setFieldValue('baseAmount', calculatedAmount)
+  }, [formik.values.isSubjectToVAT, initialVatPct, formik.values.subtotal])
+
+  useEffect(() => {
+    ;(async function () {
+      try {
+        if (recordId) {
+          const res = await getRequest({
+            extension: FinancialRepository.FiMemo.get,
+            parameters: `_recordId=${recordId}`
+          })
+
+          formik.setValues({
+            ...res.record,
+
+            date: formatDateFromApi(res.record.date)
+          })
+        }
+      } catch (exception) {}
+    })()
+  }, [])
+  console.log(formik.values, 'formike')
+  useEffect(() => {
+    formik.setFieldValue('templateId', '')
+  }, [formik.values.notes])
+
+  const onPost = async () => {
+    try {
+      const res = await postRequest({
+        extension: FinancialRepository.FiMemo.post,
+        record: JSON.stringify(formik.values)
+      })
+
+      if (res?.recordId) {
+        toast.success('Record Posted Successfully')
+        invalidate()
+
+        try {
+          const getRes = await getRequest({
+            extension: FinancialRepository.FiMemo.get,
+            parameters: `_recordId=${formik.values.recordId}`
+          })
+
+          getRes.record.date = formatDateFromApi(getRes.record.date)
+          formik.setValues(getRes.record)
+        } catch (getError) {}
+      }
+    } catch (postError) {}
+  }
+
+  const onCancel = async () => {
+    try {
+      const res = await postRequest({
+        extension: FinancialRepository.FiMemo.cancel,
+        record: JSON.stringify(formik.values)
+      })
+
+      if (res?.recordId) {
+        toast.success('Record Canceled Successfully')
+        invalidate()
+
+        try {
```

Comment:
for the whole functions if adding two try & cash in same block has no purpose then please remove them and keep one only